### PR TITLE
New offline pu

### PR DESCRIPTION
**Pull request by jbrands**
This request is for including the new offline pu Id weights.
There's a new parameter set in RecoJets/JetProducers/python/puJetIDAlgo_cff.py
I included a new variable "etaBinnedWeights" because the new weights a trained for 4 different eta regions.  
The new variables are included in RecoJets/JetProducers/src/PileupJetIdAlgo.cc

The code can be checked with RecoJets/JetProducers/test/testJetTools_cfg.py

I did the runTheMatrix.py check without error messages

Some discrimination plots can be find in:
http://www.hephy.at/user/jbrandstetter/

The pull request for the new offline weights is:
https://github.com/cms-data/RecoJets-JetProducers/pull/1
The weight file have to be taken out of my release as soon as this request is approved.

As soon as this request is approved I will continue with a request for including the hlt pu weights.

### CODE REVIEW

**Comment by slava77** [comment]
are the files  RecoJets/JetProducers/test/full_config.py and RecoJets/JetProducers/test/testJetTools.root
 needed ?
please remove

**Comment by slava77** [comment]
@cmsbuild please test

it looks like https://github.com/cms-data/RecoJets-JetProducers/pull/1 was merged
we can wait for the next IB and then this PR will need to be updated to remove the /data files

File: RecoJets/JetProducers/python/pileupjetidproducer_cfi.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
                                       runMvas = cms.bool(True),
                                       jets = cms.InputTag("ak4PFJetsCHS"),
                                       vertexes = cms.InputTag("offlinePrimaryVertices"),
```
Comment (ahinzmann) [inline RecoJets/JetProducers/python/pileupjetidproducer_cfi.py]: This config seems to be missing "etaBinnedWeights"

Comment (ahinzmann) [inline RecoJets/JetProducers/python/pileupjetidproducer_cfi.py]: In fact there is yet another file
https://github.com/jbrands/cmssw/blob/newOfflinePuID/RecoJets/JetProducers/python/PileupJetID_cfi.py
that looks like a renamed version of this config that may need this change as well.

Probably (after this PR introducing the new ID) a merge of the various cfi defining the offline pilupjet ID is needed as well:
https://github.com/jbrands/cmssw/blob/newOfflinePuID//PhysicsTools/PatAlgos/python/slimming/pileupJetId_cfi.py
https://github.com/jbrands/cmssw/blob/newOfflinePuID/RecoJets/JetProducers/python/PileupJetID_cfi.py
https://github.com/jbrands/cmssw/blob/newOfflinePuID/RecoJets/JetProducers/python/pileupjetidproducer_cfi.py

**Comment by ahinzmann** [comment]
@slava77: Could you relaunch the tests to see if it works with the latest IB?

**Comment by slava77** [comment]
-1

- Please remove  JMEAnalysis/JetToolbox 
- please remove RecoJets/JetProducers/data/TMVAClas*

we can restart the jenkns tests after that.

**Comment by slava77** [comment]
please remove 
JMEAnalysis/JetToolbox
file

**Comment by slava77** [comment]
-1
looks like the next-to-last commit broke this PR
https://github.com/jbrands/cmssw/commit/be2c7a5098c5b3dd2c23df3909309e1bc34bb7be
there are merge conflicts here now.
Please fix

**Comment by jbrands** [comment]
I close this PR and have already opened a new one with all your requested changes: https://github.com/cms-sw/cmssw/pull/10398

**Pull request closed without merging**